src/main/java/io/xol/enklume/nbt/NBTLong.java

```diff
@@ -10,13 +10,13 @@
     void feed(DataInputStream is) throws IOException {
         super.feed(is);
 
-        data = is.read() << 56;
-        data += is.read() << 48;
-        data += is.read() << 40;
-        data += is.read() << 32;
-        data += is.read() << 24;
-        data += is.read() << 16;
-        data += is.read() << 8;
+        data = (long) is.read() << 56;
```

Comment:
Was this broken then ? I guess I might have never realized/tested it

Reply:
is.read() provides an integer. This is then shifted out of scope so you always have zero as result. Haven't testet is myself though...

Reply:
I guess it depends how the type inference/checking works. I could see it working in a language if it starts with the knowledge the rhs to += must be a long, and then as you recurse into it keeps trying to have a long.

However, in Java this was clearly a bug: https://godbolt.org/z/fM5Wfe9e8